game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/general/FiringGroupSplitterGeneral.java

```diff
@@ -0,0 +1,135 @@
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ALIVE;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.AIR_FIRE_NON_SUBS;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.fire.FiringGroup;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import lombok.Value;
+import org.triplea.java.PredicateBuilder;
+import org.triplea.java.collections.CollectionUtils;
+
+/**
+ * Create firing groups with units that match {@link #firingUnitPredicate}
+ *
+ * <p>The firing groups are separated by canNotTarget, canNotBeTargetedBy, and isSuicideOnHit
+ */
+@Value(staticConstructor = "of")
+public class FiringGroupSplitterGeneral implements Function<BattleState, List<FiringGroup>> {
+
+  BattleState.Side side;
+
+  Predicate<Unit> firingUnitPredicate;
+
+  /** Name displayed in the Battle UI */
+  String groupName;
+
+  @Override
+  public List<FiringGroup> apply(final BattleState battleState) {
+    final Collection<Unit> canFire =
+        CollectionUtils.getMatches(
+            battleState.filterUnits(ACTIVE, side),
+            PredicateBuilder.of(firingUnitPredicate)
+                // Remove offense allied units if allied air can not participate
+                .andIf(
+                    side == OFFENSE
+                        && !Properties.getAlliedAirIndependent(battleState.getGameData()),
+                    Matches.unitIsOwnedBy(battleState.getPlayer(side)))
+                .build());
+
+    final Collection<Unit> enemyUnits =
+        CollectionUtils.getMatches(
+            battleState.filterUnits(ALIVE, side.getOpposite()),
+            PredicateBuilder.of(Matches.unitIsNotInfrastructure())
+                .andIf(side == DEFENSE, Matches.unitIsSuicideOnAttack().negate())
+                .andIf(side == OFFENSE, Matches.unitIsSuicideOnDefense().negate())
+                .build());
+
+    final List<FiringGroup> firingGroups = new ArrayList<>();
+
+    final List<TargetGroup> targetGroups = TargetGroup.newTargetGroups(canFire, enemyUnits);
+
+    if (targetGroups.size() == 1) {
+      firingGroups.addAll(buildFiringGroups(groupName, canFire, enemyUnits, targetGroups.get(0)));
+    } else {
+      // General firing groups don't have individual names so find commonly used groups and
+      // give them unique names
+      final List<TargetGroup> airVsSubGroups =
+          targetGroups.stream()
+              .filter(this.filterAirVsSubTargetGroups(enemyUnits))
+              .collect(Collectors.toList());
+      generateNamedGroups(AIR_FIRE_NON_SUBS, firingGroups, airVsSubGroups, canFire, enemyUnits);
```

Comment:
This special named group will replace the "Air defend non subs" or "Air attack non subs" battle steps.  I mentioned this and posted images of what it will look like in the original PR #7823